spec/plotting/dataframe_spec.rb

```diff
@@ -301,3 +301,86 @@ def == other
     it { expect { df.plot(type: :box, categorized: {by: :c, method: :color}) }.to raise_error ArgumentError }
   end
 end
+
+describe Daru::DataFrame, 'plotting dataframe using gruff' do
+  before { Daru.plotting_library = :gruff }
+  let(:df) do
+    Daru::DataFrame.new({
+      a: [1, 3, 5, 2, 5, 0],
+      b: [1, 5, 2, 5, 1, 0],
+      c: [1, 6, 7, 2, 6, 0]
+    }, index: 'a'..'f')
+  end
+
+  context 'bar' do
+    let(:plot) { instance_double 'Gruff::Bar' }
```

Comment:
`instance_double` here creates a fake instance of `Gruff::Bar` so it can take note of methods we can call on it and verify it.